vendor/bat-native-ads/src/bat/ads/internal/federated/covariate_logs.cc

```diff
@@ -18,15 +18,120 @@
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastClickedReloadButtonEvent}},
+         {UserActivityEventType::kClosedTab,
+          {brave_federated::mojom::CovariateType::kNumberOfClosedTabEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastClosedTabEvent}},
+         {UserActivityEventType::kFocusedOnExistingTab,
+          {brave_federated::mojom::CovariateType::
+               kNumberOfFocusedOnExistingTabEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastFocusedOnExistingTabEvent}},
+         {UserActivityEventType::kNewNavigation,
+          {brave_federated::mojom::CovariateType::kNumberOfNewNavigationEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastNewNavigationEvent}},
+         {UserActivityEventType::kOpenedNewTab,
+          {brave_federated::mojom::CovariateType::kNumberOfOpenedNewTabEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastOpenedNewTabEvent}},
+         {UserActivityEventType::kPlayedMedia,
+          {brave_federated::mojom::CovariateType::kNumberOfPlayedMediaEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastPlayedMediaEvent}},
+         {UserActivityEventType::kSubmittedForm,
+          {brave_federated::mojom::CovariateType::kNumberOfSubmittedFormEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastSubmittedFormEvent}},
+         {UserActivityEventType::kTypedAndSelectedNonUrl,
+          {brave_federated::mojom::CovariateType::
+               kNumberOfTypedAndSelectedNonUrlEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastTypedAndSelectedNonUrlEvent}},
+         {UserActivityEventType::kTypedKeywordOtherThanDefaultSearchProvider,
+          {brave_federated::mojom::CovariateType::
+               kNumberOfTypedKeywordOtherThanDefaultSearchProviderEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastTypedKeywordOtherThanDefaultSearchProviderEvent}},
+         {UserActivityEventType::kTypedUrl,
+          {brave_federated::mojom::CovariateType::kNumberOfTypedUrlEvents,
+           brave_federated::mojom::CovariateType::
+               kTimeSinceLastTypedUrlEvent}}});
+
+const std::vector<int> average_clickthrough_rate_time_windows({1, 7, 28});
+
+}  // namespace
+
 CovariateLogs::CovariateLogs() {
-  SetCovariateLogEntry(
-      std::make_unique<AdNotificationNumberOfTabsOpenedInPast30Minutes>());
   SetCovariateLogEntry(
       std::make_unique<
           AdNotificationLocaleCountryAtTimeOfServingCovariateLogEntry>());
+  SetCovariateLogEntry(
+      std::make_unique<AdNotificationNumberOfTabsOpenedInPast30Minutes>());
+  SetCovariateLogEntry(std::make_unique<LastAdNotificationWasClicked>());
+
+  for (const auto& time_window_in_days :
```

Comment:
nit: `average_clickthrough_rate_time_window` if we want to add `_in_days` I think we should also add this to `average_clickthrough_rate_time_windows`